data/config-schema/src/main/resources/net/corda/schema/configuration/stateManager/1.0/stateManager-db-properties.json

```diff
@@ -1,101 +1,125 @@
 {
   "$schema": "https://json-schema.org/draft/2019-09/schema",
   "$id": "https://corda.r3.com/net/corda/schema/configuration/stateManager/1.0/stateManager-db-properties.json",
-  "title": "State Manager Database Configuration Schema",
-  "description": "Configuration schema for the database section. Note that this configuration cannot be updated dynamically through the REST endpoint.",
+  "title": "State Manager configuration schema for database implementation",
+  "description": "Configuration schema for State Manager's database implementation. Note that this configuration cannot be updated dynamically through the REST endpoint.",
   "type": "object",
-  "properties": {
-    "database": {
-      "description": "Database settings.",
-      "type": "object",
-      "default": {},
-      "properties": {
-        "user": {
-          "description": "The database user.",
-          "type": "string"
-        },
-        "pass": {
-          "description": "The database password.",
-          "type": "string"
-        },
-        "jdbc": {
-          "description": "JDBC settings.",
-          "type": "object",
-          "default": {},
-          "properties": {
-            "url": {
-              "description": "The JDBC URL.",
-              "type": "string",
-              "default": "jdbc:postgresql://state-manager-db:5432/state_manager"
-            },
-            "driver": {
-              "description": "The JDBC driver.",
-              "type": "string",
-              "default": "org.postgresql.Driver"
+  "additionalProperties": {
```

Comment:
I don't think this one is needed here?. Once removed, also, the diff should be become way smaller.

Reply:
This seems to be how to represent a "map" in json schema, the keys in this map are the state type, for example, the following map of state types json matches the schema:

```
{
  "flowCheckpoint": {
    "database": {
      "jdbc": {
        "driver": "org.postgresql.Driver",
        "url": "1url"
      },
      "password": "1pass",
      "connectionPool": {
        "idleTimeoutSeconds": 120000000,
        "keepAliveTimeSeconds": 0,
        "maxLifetimeSeconds": 1800000000,
        "maxSize": 5,
        "minSize": 1,
        "validationTimeoutSeconds": 5000000
      },
      "username": "1user"
    },
    "type": "DATABASE"
  },
  "flowMapping": {
    "database": {
      "jdbc": {
        "driver": "org.postgresql.Driver",
        "url": "1url"
      },
      "password": "1pass",
      "connectionPool": {
        "idleTimeoutSeconds": 120000000,
        "keepAliveTimeSeconds": 0,
        "maxLifetimeSeconds": 1800000000,
        "maxSize": 5,
        "minSize": 1,
        "validationTimeoutSeconds": 5000000
      },
      "username": "1user"
    },
    "type": "DATABASE"
  },
  "keyRotation": {
    "database": {
      "jdbc": {
        "driver": "org.postgresql.Driver",
        "url": "1url"
      },
      "password": "1pass",
      "connectionPool": {
        "idleTimeoutSeconds": 120000000,
        "keepAliveTimeSeconds": 0,
        "maxLifetimeSeconds": 1800000000,
        "maxSize": 5,
        "minSize": 1,
        "validationTimeoutSeconds": 5000000
      },
      "username": "1user"
    },
    "type": "DATABASE"
  },
  "p2pSession": {
    "database": {
      "jdbc": {
        "driver": "org.postgresql.Driver",
        "url": "1url"
      },
      "password": "1pass",
      "connectionPool": {
        "idleTimeoutSeconds": 120000000,
        "keepAliveTimeSeconds": 0,
        "maxLifetimeSeconds": 1800000000,
        "maxSize": 5,
        "minSize": 1,
        "validationTimeoutSeconds": 5000000
      },
      "username": "1user"
    },
    "type": "DATABASE"
  },
  "tokenPoolCache": {
    "database": {
      "jdbc": {
        "driver": "org.postgresql.Driver",
        "url": "1url"
      },
      "password": "1pass",
      "connectionPool": {
        "idleTimeoutSeconds": 120000000,
        "keepAliveTimeSeconds": 0,
        "maxLifetimeSeconds": 1800000000,
        "maxSize": 5,
        "minSize": 1,
        "validationTimeoutSeconds": 5000000
      },
      "username": "1user"
    },
    "type": "DATABASE"
  }
}
```